backend/app/controllers/api/v1/roles_controller.rb

```diff
@@ -2,6 +2,12 @@ class Api::V1::RolesController < Api::V1::ApiController
   skip_before_action :authenticate_user!, only: [:index]
 
   def index
-    render json: Role.all
+    roles = if current_user&.admin?
```

Comment:
is not good to have queries like this here, can you think about something better to use?